In [7]:
import pandas as pd
from pandas import DataFrame

**Adding a column mapping library in order to make columns easier to work with:**

In [10]:
column_mapping = {
    'Stn Id': 'station_id',
    'Stn Name': 'station_name',
    'CIMIS Region': 'region',
    'Date': 'date',
    'Jul': 'jul',
    'ETo (in)': 'eto_in',
    'Precip (in)': 'precip',
    'Sol Rad (Ly/day)': 'sol_rad',
    'Avg Vap Pres (mBars)': 'avg_vap_pressure',
    'Max Air Temp (F)': 'max_air_temp',
    'Min Air Temp (F)': 'min_air_temp',
    'Min Air Temp (F)': 'min_air_temp',
    'Avg Air Temp (F)': 'avg_air_temp',
    'Max Rel Hum (%)': 'max_rel_humidity',
    'Min Rel Hum (%)': 'min_rel_humidity',
    'Avg Rel Hum (%)': 'avg_rel_humidity',
    'Dew Point (F)': 'dew_point',
    'Avg Wind Speed (mph)': 'avg_wind_speed',
    'Wind Run (miles)': 'wind_run_miles',
    'Avg Soil Temp (F)': 'avg_soil_temp',

}
    

---
*read in CVS file and apply mapping*

In [11]:
weather = pd.read_csv('daily-Copy1.csv')
weather = weather.rename(columns=column_mapping)

In [18]:
print('Date Range: ', weather['date'].min(), 'to', weather['date'].max())
print('Hottest Day: ', weather.loc[weather['max_air_temp'].idxmax(), ['date', 'max_air_temp']])
print('Coolest Day: ', weather.loc[weather['min_air_temp'].idxmin(), ['date', 'min_air_temp']])

Date Range:  5/10/2025 to 8/5/2025
Hottest Day:  date            7/13/2025
max_air_temp         97.7
Name: 69, dtype: object
Coolest Day:  date            6/15/2025
min_air_temp         28.1
Name: 41, dtype: object


Dig deeper into the 28.1 degree day on June 15


In [20]:
cold_day = weather[weather['date'] == '6/15/2025']
print(cold_day)

    station_id          station_name                        region       date  \
41         264  Sierra Valley Center  Eastern Sierra Basin & Range  6/15/2025   

    jul  eto_in qc  precip qc.1  sol_rad  ... avg_rel_humidity  qc.9  \
41  166    0.25        0.0         665.0  ...             41.0         

   dew_point  qc.10 avg_wind_speed  qc.11 wind_run_miles  qc.12 avg_soil_temp  \
41      33.8                   4.5                 108.9                 59.5   

    qc.13  
41         

[1 rows x 33 columns]


---
the QC flag was set on 6/15/2025, false alarm, no metereological anomoly, sensor error
